package.json

```diff
@@ -19,9 +19,10 @@
   },
   "homepage": "https://github.com/DEFRA/ffc-pay-event-publisher#readme",
   "dependencies": {
-    "joi": "^17.4.2"
+    "joi": "17.4.2",
+    "uuid": "9.0.0"
   },
   "devDependencies": {
-    "standard": "^16.0.4"
```

Comment:
Just curious, but was just wondering what the ^ means here?

Reply:
It's a way of doing fuzzy version matching.

So with the `^` as a prefix, when you run `npm install` it will install the latest minor/patch version for that major version.

For example, this is set to `^16.0.4`.  So if say there was a later version of `16.1.0` that would get installed instead.  However, if there was also a `17.0.0` that wouldn't be because it's a major version change.

You can also do `~16.0.4`  which means that only patch version updates are allowed.  For example, would be able to install `16.0.5` but not `16.1.0`.

We agreed not to use the `^` or `~` because it adds risk to delivery.  For example, if locally I build a container and run and test against one version of a dependency.  It might not be the same version that Jenkins installs when it builds and publishes the container for our Azure environments.  So the risk is the behaviour may be different which we don't want.

So safest thing to do is just say "I want exactly this version".

By default when you install a new package it will add the `^`.  Hence why a few have creeped into our repos. But you can create a .npmrc file in your home directory with this content to change that behaviour.

![image](https://user-images.githubusercontent.com/42574977/224292466-fc7cdc93-f395-4777-b883-7cd255e8e551.png)

Reply:
Ah okay, that clears that up. Thanks! Handy information to know :)